### PR TITLE
modify entropy

### LINKED ISSUE #112: Write Entropy Tests and Entropy code is not right
## Check list 
#99

### PR DESCRIPTION
**Pull request by kaikai529**
## Description

modify entropy
## Related Issues and Reference

Close #112 
## Test

No

### CODE REVIEW

**Comment by Hattomo** [comment]
https://github.com/Hattomo/kaikai_AI/pull/124#issuecomment-726818459
- [x] entropy
- [x] diff entropy

File: shared/costfunction.py [hunk 1]
```diff
@@ -13,20 +14,20 @@ def rss(label, ans):
 def diffrss(label, ans):
     return ans - label
 
+def non_universal_cross_entropy(label, ans):
+    if ans < 0:
+        sys.stdout.write(
+            "Calculation Error(cross entropy): The actfunc is not right.\nplease change actfunc of return only plus\n")
+        sys.exit(1)
+    return -label * math.log(ans) - (1-label) * math.log(1 - ans)
+
```
Comment (Hattomo) [inline shared/costfunction.py]: Use numpy?

**Comment by kaikai529** [comment]
add test

**Review by kaikai529 — COMMENTED** [review]
I refactor entropy and please check it again

File: tests/test_costfunction.py [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
\ No newline at end of file
+import costfunction
+
+def setUp():
+    # init
+    pass
+
+def tearDown():
+    # dispose
+    pass
```
Comment (Hattomo) [inline tests/test_costfunction.py]:
```suggestion

```

File: tests/test_costfunction.py [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
+
+def test_rss():
+    pass
+
+def test_cross_entropy():
+    label = np.array([0, 0, 1, 1])
+    ans = np.array([0.2, 0.3, 0.4, 0.5])
+    error = costfunction.cross_entropy(label, ans)
+    assert error - 2.189 < 1e-3, "check cross_entropy or test"
```
Comment (Hattomo) [inline tests/test_costfunction.py]: 1. Can you divide into to two tests, `test_cross_entropy()` and `test_diff_cross_entropy()`?


2. Please Use `np.isclose()`.
```
# This passed test
assert error - 2000000.189 < 1e-3, "check cross_entropy or test"
```

**Review by Hattomo — CHANGES_REQUESTED** [review]
Great Works!

File: shared/costfunction.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import math
```
Comment (Hattomo) [inline shared/costfunction.py]:
```suggestion

```

File: tests/test_costfunction.py (hunk 3 above)
Comment (kaikai529) [reply tests/test_costfunction.py]: It's my miss so modify this immediately.
1 yes
2 yes

**Review by Hattomo — APPROVED** [review]
@kaikai529 Sorry, we should use `allcolse()` than `isclose()`. I'll merge because this tests work well.

**Pull request merged**